include/library.h

```diff
@@ -263,6 +263,18 @@ class Library: public std::enable_shared_from_this<Library>
    */
   bool removeBookmark(const std::string& zimId, const std::string& url);
 
+  /**
+   * Resolve bookmarks
+   *
+   * @param source the source bookId of the bookmarks to resolve.
+                   If not present, we resolve all bookmarks pointing to an unknown book.
+   * @param destination the destination bookId to resolve te bookmark to.
+                        If not present, we will try to determine it using heuristics based on
+                        book name, flavour and date.
+   * @return The number of bookmarks created.
+   */
+  int resolveBookmarks(const std::string& source = "", std::string destination = "");
```

Comment:
I think it's better to have three separate functions:

```
migrateBookmarks(source, destination);
migrateBookmarks(source);
migrateBookmarks();
```
where `source` and `destination` must be valid book-ids. This eliminates the meaningless combination `migrateBookmarks("", destination)` and decomposes the implementation into a set of smaller and easier to understand non-recursive functions.